src/Profiles.ts

```diff
@@ -1091,6 +1126,70 @@ export class Profiles {
         return filteredProfile;
     }
 
+    public async getCombinedProfile(serviceProfile: IProfileLoaded, baseProfile: IProfileLoaded) {
+
+        // TODO: This needs to be improved
+        // The idea is to handle all type of ZE Profiles
+
+        // This check will handle service profiles that hase username and password
+        if (serviceProfile.profile.user && serviceProfile.profile.password) {
+            return serviceProfile;
+        }
+
+        // This check is for optional credentials
+        if (baseProfile && serviceProfile.profile.host && serviceProfile.profile.port &&
+            ((baseProfile.profile.host !== serviceProfile.profile.host) && baseProfile.profile.port !== serviceProfile.profile.port)) {
+                return serviceProfile;
+        }
+
+        // This process combines the information from baseprofile to serviceprofile and create a new session
+        const cmdArgs: ICommandArguments = {
+            $0: "zowe",
+            _: [""],
+            host: serviceProfile.profile.host ? serviceProfile.profile.host :
+            (baseProfile ? baseProfile.profile.host : undefined),
+            port: serviceProfile.profile.port ? serviceProfile.profile.port :
+            (baseProfile ? baseProfile.profile.port : 0),
+            basePath: serviceProfile.profile.basePath ? serviceProfile.profile.basePath :
+            (baseProfile ? baseProfile.profile.basePath : undefined),
+            rejectUnauthorized: serviceProfile.profile.rejectUnauthorized !== null ?
+            serviceProfile.profile.rejectUnauthorized :
+            (baseProfile ? baseProfile.profile.rejectUnauthorized : true),
+            user: serviceProfile.profile.user ? serviceProfile.profile.user :
+            (baseProfile ? baseProfile.profile.user : undefined),
+            password: serviceProfile.profile.password ? serviceProfile.profile.password :
+            (baseProfile ? baseProfile.profile.password : undefined),
+            tokenType: baseProfile.profile.tokenType,
+            tokenValue: (baseProfile && !serviceProfile.profile.password) ? baseProfile.profile.tokenValue : undefined
+        };
+
+        const combinedSession = await zowe.ZosmfSession.createBasicZosmfSessionFromArguments(cmdArgs);
```

Comment:
This is zosmf specific. We proposed in our branch yesterday a new Common API called `getSessionFromCommandArgument?(cmdArgs: ICommandArguments): Session;`  This will also break extenders, but this code is only called if the user actually has a base profile. So users without base profiles are not broken; not are zosmf users.  We could ship new versions of our extensions that then explicitly add base profile support. 

Also the else branch when we test for the existence of `commonApi.getSessionFromCommandArgument` could call the zosmf implementation (also see our branch for the zosmf implementation of this new common api, which that line of code again, but inside the zosmf api implementation module.

Reply:
I thinkkk I got this one, I copied over `getSessionFromCommandArgument` from your branch into the API file and am calling it if the `commonApi` doesn't have method `getSessionFromCommandArgument` Please check if it is okay though.